### ISSUE TITLE
normalization error in render_gaussian2d

### ISSUE THREAD

**Issue by themattinthehatt** [kornia/kornia#1710]
### Describe the bug

In the source code for kornia.geometry.subpix.render_gaussian2d, there is a normalization step where the heatmap should be divided by sigma squared, but instead is divided by sigma:
```
# ks <- -1 / (2 \sigma^2)
k_x = -0.5 * torch.reciprocal(std[..., 0, None, None])
k_y = -0.5 * torch.reciprocal(std[..., 1, None, None])
```
https://github.com/kornia/kornia/blob/44c97a0a9957957559d0a8c9225be67a89306778/kornia/geometry/subpix/dsnt.py#L139

I believe this should instead be
```
# ks <- -1 / (2 \sigma^2)
k_x = -0.5 * torch.reciprocal(torch.square(std[..., 0, None, None]))
k_y = -0.5 * torch.reciprocal(torch.square(std[..., 1, None, None]))
```

### Reproduction steps

```bash
Math bug, no reproduction code necessary
```

### Expected behavior

Return a properly normalized 2d Gaussian; currently returns a heatmap that sums to one but is of the wrong mathematical form.

### Environment

```shell
n/a
```

**Comment by themattinthehatt** [comment]
reopening with a new "bug" label

**Comment by edgarriba** [comment]
@themattinthehatt thanks to point this out. Would you mind to file a quick PR with the fix ? /cc @anibali

**Comment by themattinthehatt** [comment]
Sure thing, traveling now but will submit in a day or two

**Comment by juliaaz** [comment]
Hi @themattinthehatt @edgarriba I see that this issue is still not fixed. I just changed the formula and I am trying to fix the tests, however i see that in tests matrix is hardcoded. Do I need then to change the values of it or i can just do the following:

`    @pytest.fixture()
    def gaussian(self, device, dtype):
        mean = torch.tensor([0.0, 0.0], dtype=dtype, device=device)
        std = torch.tensor([0.25, 0.25], dtype=dtype, device=device)
        return kornia.geometry.subpix.render_gaussian2d(mean, std, (5, 5), True)
    
    @pytest.fixture()
    def gaussian_pixel_coordinates(self, device, dtype):
        mean = torch.tensor([2.0, 2.0], dtype=dtype, device=device)
        std = torch.tensor([1.0, 1.0], dtype=dtype, device=device)
        return kornia.geometry.subpix.render_gaussian2d(mean, std, (5, 5), False)`

**Comment by themattinthehatt** [comment]
@juliaaz I think the hard-coding of the means and stds shouldn't be an issue here. The question is where these fixtures get used in other tests. Since the standard deviation computation is changed downstream tests might fail (or maybe not, depends on the test coverage). One simple thing to try would be to fix the bug, then run the tests and see what fails.
